xarray/tests/test_dataset.py

```diff
@@ -5956,6 +5956,8 @@ def test_coarsen_keep_attrs():
         attrs=_attrs,
     )
 
+    ds2 = ds.copy(deep=True)
+
     # Test dropped attrs
     dat = ds.coarsen(coord=5).mean()
```

Comment:
can we add a test to `test_dataarray` too please?

Reply:
There are currently no tests on coarsen in test_dataarray. Would you like me to add tests similar to all of the coarsen tests in test_variable.py -- or just for maintaining the original object -  at least at this stage?

Reply:
I added a test and coarsen is failing to drop the attributes of a dataarray. It might take some time to understand why this is.